### DISCUSSION TITLE
searchAll: False

### DISCUSSION THREAD

**Discussion by Vyerni** [cross-seed/cross-seed#91] [Q&A] [answered]
Just trying to understand what exactly changes between setting the search all option from false to true in the config file?

I understand it overrides the single episode option. But other's have been telling me that with it set to false, it doesn't search for full discs either. 

Anything else that it misses? Just wondering whether it's worth me running it as true, and letting it run through single episodes?

**Comment by mmgoodnow** [comment, marked as answer]
There are (as of right now) two prefilter criteria (I call it content-based filtering, as opposed to timestamp filtering or dupe filtering) that searchAll disables:
- single episodes
- torrents in which not all listed files are video files

The other prefiltering stages are torrent deduplication and timestamp filtering, if you're using either of the timestamp flags (--exclude-recent-search and --exclude-older). Neither of these stages are affected by the searchAll flag.

**Reply by Vyerni** [reply]
Just wondering if it'd be possible for the search all to be a different function compared to the single episode? 

So that I could run search all, so it would work for full discs/torrents with NFO's/Pictures in them, but it wouldn't bother searching individual episodes too?

**Reply by mmgoodnow** [reply]
You can file an issue for it, but it'll be pretty low priority. I will get to it anywhere from several months from now to never. If you have programming experience, your best bet is to fork the repo and manually remove the second prefilter criterion. I would accept a pull request for it if you're willing to turn it into a command line option and do the additional documentation work. 

If you don't have programming experience you're best off finding somebody who does and cares about this use case. 

> so it would work for full discs/torrents with NFO's/Pictures in them

At the beginning I made this choice because I figured that most torrents containing NFOs or pictures wouldn't be able to cross-seed anyways, because the extra stuff is more tracker-specific. Do you have a sense of how many potential matches you're missing out on due to this prefiltering criterion? This assumption of mine may have been incorrect.

### tangent

More generally, it's clear that people want to be able to do fuzzier matching, and I understand that, but cross-seed was originally developed as a tool for myself, and I have no desire to squeeze the last ounce of juice out of the proverbial lemon. As every OSS dev knows, maintainer burnout is a very real problem, and it's a lot harder to motivate myself when there's no personal gain. 

cross-seed's strictness has a couple benefits, mainly

- you don't have to deal with false positives
- I don't have to deal with false positives (in the code)

as long as it's working correctly.

Allowing fuzzier matching will require a lot more false positive handling, testing in clients I don't use / am unfamiliar with (looking at you qbit), and thought on how I want it to work. It'll probably come eventually but definitely don't hold your breath.

**Reply by Vyerni** [reply]
Nah, I have no coding experience. But I'm more than happy to run it with search all set as true, and then just delete the individual episode torrents it grabs. Just nit-picking really, because it's an absolutely lovely tool that you've developed anyway.
